unblob/handlers/archive/tar.py

```diff
@@ -146,3 +140,70 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
         if end_offset == -1:
             return None
         return ValidChunk(start_offset=start_offset, end_offset=end_offset)
+
+
+class TarUstarHandler(_TarHandler):
+    PATTERNS = [
+        HexString("75 73 74 61 72 20 20 00"),
+        HexString("75 73 74 61 72 00 30 30"),
+    ]
+
+    # Since the magic is at 257, we have to subtract that from the match offset
+    # to get to the start of the file.
+    PATTERN_MATCH_OFFSET = -MAGIC_OFFSET
+
+
+def _re_protect(regexp: str):
+    return f"({regexp})"
+
+
+def _re_alternatives(regexps):
+    return _re_protect("|".join(_re_protect(regexp) for regexp in regexps))
```

Comment:
From the name `_re_protect` I have not associated to capture groups, but quoting or something similar. I'd rename this to `_re_group` instead.

Also, why do you need grouping the inner alternatives?

If I am not mistaken the output of this is:
`((re1)|(re2)|...|(reN))` Why doesn't `(re1|re2|...|reN)` suffice?

Reply:
Protecting the regular expression from appending/prefixing is the intention here, grouping is an implementation detail, so the name is right.

The internal regexps might not need to be grouped, but I am always unsure about precedences of regexp constructs (especially `|`), this is the reason. It should not hurt either, as hyperscan supports only non-capture groups.
So while it could be unneeded, I'd prefer it this way.

Reply:
This should be made obvious somehow. I have not figured out this intent. At least an explanatory comment would be welcome.

Reply:
Renamed the function and added a doc-string explaining this, thanks for the question and suggestion, this had become better because of those.